src/cephadm/cephadmlib/daemons: Fix typos in ceph.py and elsewhere

anthonyeleven opened the pull request:
src/cephadm/cephadmlib/daemons: Fix typos in ceph.py and Dashboard files.

Pointed out by Eugen Block.
Fixes: https://tracker.ceph.com/issues/64242
Signed-off-by: Anthony D'Atri <anthonyeleven@users.noreply.github.com>
<!--
  - Please give your pull request a title like

      [component]: [short description]

  - Please use this format for each git commit message:

      [component]: [short description]

      [A longer multiline description]

      Fixes: [ticket URL on tracker.ceph.com, create one if necessary]
      Signed-off-by: [Your Name] <[your email]>

    For examples, use "git log".
-->

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [x] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins test classic perf` [Jenkins Job](https://jenkins.ceph.com/view/all/job/ceph-perf-classic/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-perf-pull-requests/config/definitions/ceph-perf-pull-requests.yml)
- `jenkins test crimson perf` [Jenkins Job](https://jenkins.ceph.com/view/all/job/ceph-perf-crimson/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-perf-pull-requests/config/definitions/ceph-perf-pull-requests.yml)
- `jenkins test signed` [Jenkins Job](https://jenkins.ceph.com/job/ceph-pr-commits/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-pr-commits/config/definitions/ceph-pr-commits.yml)
- `jenkins test make check` [Jenkins Job](https://jenkins.ceph.com/job/ceph-pull-requests/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-pull-requests/config/definitions/ceph-pull-requests.yml)
- `jenkins test make check arm64` [Jenkins Job](https://jenkins.ceph.com/job/ceph-pull-requests-arm64/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-pull-requests-arm64/config/definitions/ceph-pull-requests-arm64.yml)
- `jenkins test submodules` [Jenkins Job](https://jenkins.ceph.com/view/all/job/ceph-pr-submodules/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-pr-submodules/config/definitions/ceph-pr-commits.yml)
- `jenkins test dashboard` [Jenkins Job](https://jenkins.ceph.com/view/all/job/ceph-dashboard-pull-requests/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-dashboard-pull-requests/config/definitions/ceph-dashboard-pull-requests.yml)
- `jenkins test dashboard cephadm` [Jenkins Job](https://jenkins.ceph.com/view/all/job/ceph-dashboard-cephadm-e2e/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-dashboard-cephadm-e2e/config/definitions/ceph-dashboard-cephadm-e2e.yml)
- `jenkins test api` [Jenkins Job](https://jenkins.ceph.com/view/all/job/ceph-api/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-pr-api/config/definitions/ceph-pr-api.yml)
- `jenkins test docs` [ReadTheDocs](https://readthedocs.org/projects/ceph/) | [Github Workflow Definition](https://github.com/ceph/ceph/blob/main/.readthedocs.yml)
- `jenkins test ceph-volume all` [Jenkins Jobs](https://jenkins.ceph.com/view/ceph-volume%20PR/) | [Jenkins Jobs Definition](https://github.com/ceph/ceph-build/blob/main/ceph-volume-cephadm-prs/config/definitions/ceph-volume-pr.yml)
- `jenkins test windows` [Jenkins Job](https://jenkins.ceph.com/job/ceph-windows-pull-requests/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-windows-pull-requests/config/definitions/ceph-windows-pull-requests.yml)
- `jenkins test rook e2e` [Jenkins Job](https://jenkins.ceph.com/view/all/job/ceph-orchestrator-rook-e2e/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-rook-e2e/config/definitions/ceph-orchestrator-rook-e2e.yml)
</details>

anthonyeleven commented:
jenkins test windows
jenkins test dashboard

phlogistonjohn commented:
What's going on here? This seems to be opened and closed multiple times.

Also, has this PR grown a bunch of stuff since I last looked at it?  I only remember seeing the one 'directory' typo and approving that and *that alone*.

phlogistonjohn reviewed (commented):
As long as this PR remains open, my suggestion is to split the commits up by file. Since these are spelling fixes they should largely be independent of each other (one exception would be if a test case made use of a misspelled string). 

Also please update the title of the PR.

anthonyeleven commented:
Closing and reopening yet again to try to get checks to pass.

phlogistonjohn commented:
> > What's going on here? This seems to be opened and closed multiple times.
> 
> That's the only way to retrigger the checks that randomly fail.
> 

Maybe it's a different set of tests for me but I usually have success with `jenkins test ...`. 
As a "build sig" member I feel duty bound to find out more about this workflow issue you're having. Could you share more info?

> > Also, has this PR grown a bunch of stuff since I last looked at it? I only remember seeing the one 'directory' typo and approving that and _that alone_.
> 
> Yes there were other typos brought to my attention. Rather than trying to juggle a zillion PRs trying to get each to pass the checks and approval, and the tribal process of Teuthology validation, I added them here. I won't merge until such a time that someone approves them.

Fine, but we really should split them on a patch-by-patch basis then.

`the tribal process of Teuthology validation`  hehe.  Unfortunately this is a double edged sword. Involve too many components in one PR and nobody will want to review/run it through teuthology.





> > As long as this PR remains open, my suggestion is to split the commits up by file.
> 
> I'm not a git wizard. My options are:
> 
> a) Amend b) Don't try to fix at all.
> 

Are you open to suggestions?

```
# "undo" the single commit you have
git reset HEAD^
# see the changed files
git status
# ... changed files should be listed here ...

# commit file-by-file
git add <filename>
git commit [-s]
```

You can also replace the iterated `git add foo` & `git commit` commands with `git commit [-s] -p` the latter will prompt for what files (and parts of those files) to include in the new commit. It's my personally preferred approach.



> > Since these are spelling fixes they should largely be independent of each other (one exception would be if a test case made use of a misspelled string).
> > Agreed.
> 
> > Also please update the title of the PR.
> Could swear I had.


:shrug: I see `src/cephadm/cephadmlib/daemons: Fix typos in ceph.py and elsewhere` so I guess "and elsewhere" is carrying a lot of weight. I didn't even notice it until I started writing this response. My mind places a lot of weight on the topic/prefix  (the part before the first colon) :-)

anthonyeleven commented:
> > > What's going on here? This seems to be opened and closed multiple times.
> > 
> > 
> > That's the only way to retrigger the checks that randomly fail.
> 
> Maybe it's a different set of tests for me but I usually have success with `jenkins test ...`.

Please note above where my attempts there elicited no reply from Jenkins.  Over time they have done so on at most 20% of my attempts.

>As a "build sig" member I feel duty bound to find out more about this workflow issue you're having. Could you share more info?

I don't have any info other than the history of failed checks in this and other PRs.  I've pasted some of the excerpts into Slack:

* In one case, a single check for a PR had claimed to be running since *July*
* Today at least one of my open PRs logged a 403 error without any context
* Most recently, complaints about apt sources not being reachable.

> > Yes there were other typos brought to my attention. Rather than trying to juggle a zillion PRs trying to get each to pass the checks and approval, and the tribal process of Teuthology validation, I added them here. I won't merge until such a time that someone approves them.
> 
> Fine, but we really should split them on a patch-by-patch basis then.

How?

> 
> `the tribal process of Teuthology validation` hehe. Unfortunately this is a double edged sword. Involve too many components in one PR and nobody will want to review/run it through teuthology.

Nobody seemed to run it through even with a single change.  It is still unclear to me how to get a PR run through Teuthology, and how to tell when it has passed and may be merged.

> > > As long as this PR remains open, my suggestion is to split the commits up by file.

Again, how?

> > I'm not a git wizard. My options are:
> > a) Amend b) Don't try to fix at all.
> 
> Are you open to suggestions?

Always.  In practical terms, please, vs a return to the hell that was my college graph theory class.

> 
> ```
> # "undo" the single commit you have
> git reset HEAD^
> # see the changed files
> git status
> # ... changed files should be listed here ...
> 
> # commit file-by-file
> git add <filename>
> git commit [-s]
> ```
> 
> You can also replace the iterated `git add foo` & `git commit` commands with `git commit [-s] -p` the latter will prompt for what files (and parts of those files) to include in the new commit. It's my personally preferred approach.

`-s` mostly hasn't worked for me in the last year.  I have to manually paste in a signed-off-by line.

But say I do the above, then others assert that the commits ust be squashed before merging -- and the auto-squash button remains disabled for our github project.

> > > Since these are spelling fixes they should largely be independent of each other (one exception would be if a test case made use of a misspelled string).

Indeed I've run into that dynamic before, e.g. when I endeavored to fix a Grafana panel whose label did not reflect its query.  I do a find/xargs grep pass through the source tree looking for such cases.

> > > Agreed.
> > 
> > 
> > > Also please update the title of the PR.
> > > Could swear I had.
> 
> 🤷 I see `src/cephadm/cephadmlib/daemons: Fix typos in ceph.py and elsewhere` so I guess "and elsewhere" is carrying a lot of weight. I didn't even notice it until I started writing this response. My mind places a lot of weight on the topic/prefix (the part before the first colon) :-)

Ack. I updated it not long ago.  In docland the convention has been that preceding the colon is the monophyletic ancestor directory that includes all changes.  The src tree seems to have different tradition.

phlogistonjohn commented:
> > > > What's going on here? This seems to be opened and closed multiple times.
> > > 
> > > 
> > > That's the only way to retrigger the checks that randomly fail.
> > 
> > 
> > Maybe it's a different set of tests for me but I usually have success with `jenkins test ...`.
> 
> Please note above where my attempts there elicited no reply from Jenkins. Over time they have done so on at most 20% of my attempts.
> 

I've seen that happen on occasion but not consistently. 20% certainly sounds very high.

> > As a "build sig" member I feel duty bound to find out more about this workflow issue you're having. Could you share more info?
> 
> I don't have any info other than the history of failed checks in this and other PRs. I've pasted some of the excerpts into Slack:
> 
>     * In one case, a single check for a PR had claimed to be running since _July_
> 
>     * Today at least one of my open PRs logged a 403 error without any context
> 
>     * Most recently, complaints about apt sources not being reachable.
> 

The first two sound like problems with jenkins and I wonder if it's overloaded.

The last one should be something that occurs "during" the build and would hopefully not prevent the normal workflow from behaving.

I'll pass the word along.


> 
> > > Yes there were other typos brought to my attention. Rather than trying to juggle a zillion PRs trying to get each to pass the checks and approval, and the tribal process of Teuthology validation, I added them here. I won't merge until such a time that someone approves them.
> > 
> > 
> > Fine, but we really should split them on a patch-by-patch basis then.
> 
> How?
> 
> > `the tribal process of Teuthology validation` hehe. Unfortunately this is a double edged sword. Involve too many components in one PR and nobody will want to review/run it through teuthology.
> 
> Nobody seemed to run it through even with a single change. It is still unclear to me how to get a PR run through Teuthology, and how to tell when it has passed and may be merged.
> 

Message received. It's been discussed in the infrastructure meeting and csc but no agreement has been reached yet. I agree right now it's a bit of a mysterious process.  Since this touches cephadm I volunteer @adk3798 and myself.

> > > > As long as this PR remains open, my suggestion is to split the commits up by file.
> 
> Again, how?
> 
> > > I'm not a git wizard. My options are:
> > > a) Amend b) Don't try to fix at all.
> > 
> > 
> > Are you open to suggestions?
> 
> Always. In practical terms, please, vs a return to the hell that was my college graph theory class.
> 
> > ```
> > # "undo" the single commit you have
> > git reset HEAD^
> > # see the changed files
> > git status
> > # ... changed files should be listed here ...
> > 
> > # commit file-by-file
> > git add <filename>
> > git commit [-s]
> > ```
> > 

 
> >   
> > You can also replace the iterated `git add foo` & `git commit` commands with `git commit [-s] -p` the latter will prompt for what files (and parts of those files) to include in the new commit. It's my personally preferred approach.
> 
> `-s` mostly hasn't worked for me in the last year. I have to manually paste in a signed-off-by line.
> 

We can visit the `-s` not working thing later.


> But say I do the above, then others assert that the commits ust be squashed before merging -- and the auto-squash button remains disabled for our github project.
> 

Who? Let me 'attem.  :-D
More seriously, if you're contributing to ceph orch I think most of the team is in favor of split commits. If some other subsystem isn't, that's their preference but we should at least be split across subsystems.
Otherwise, point me at one of these PRs next time you have this discussion and I'll try to help.

> > > > Since these are spelling fixes they should largely be independent of each other (one exception would be if a test case made use of a misspelled string).
> 
> Indeed I've run into that dynamic before, e.g. when I endeavored to fix a Grafana panel whose label did not reflect its query. I do a find/xargs grep pass through the source tree looking for such cases.
> 
> > > > Agreed.
> > > 
> > > 
> > > > Also please update the title of the PR.
> > > > Could swear I had.
> > 
> > 
> > 🤷 I see `src/cephadm/cephadmlib/daemons: Fix typos in ceph.py and elsewhere` so I guess "and elsewhere" is carrying a lot of weight. I didn't even notice it until I started writing this response. My mind places a lot of weight on the topic/prefix (the part before the first colon) :-)
> 
> Ack. I updated it not long ago. In docland the convention has been that preceding the colon is the monophyletic ancestor directory that includes all changes. The src tree seems to have different tradition.

My thesaurus broke. I will try again later. ;-)

anthonyeleven commented:
jenkins test dashboard cephadm

shraddhaag commented:
RADOS Approved: https://tracker.ceph.com/projects/rados/wiki/MAIN#httpstrackercephcomissues73052

adk3798 commented:
https://pulpito.ceph.com/adking-2025-09-17_13:26:44-orch:cephadm-wip-adk-testing-2025-09-16-1923-distro-default-smithi/

nothing to block merging from orchestration side
